meow/user_profile/views.py

```diff
@@ -5,29 +5,112 @@
+        if serializer.is_valid():
+            serializer.save()
+        else:
+            print(serializer.errors)
+            return JsonResponse(serializer.errors,status=400, safe=False)
+        return JsonResponse(id, safe=False, status=200)
+    
+    def delete(self, request, id, format=None):
+        user=request.user
+        themetodelete = Theme.objects.get(pk=id)
+        delete_name=themetodelete.name
+        if(delete_name == 'Daily Bruin' or delete_name == 'Dark Bruin'):
+            return JsonResponse('Default themes cannot be deleted', safe=False, status=400)
+        if user.selected_theme == Theme.objects.filter(name=delete_name, author=user)[0]:
+            user.selected_theme = Theme.objects.get(name="Daily Bruin")
+            user.save()
+        if themetodelete.user_set.count()==0 and themetodelete.related_users.count()==0:
+            themetodelete.delete()
+        else:
+            themetodelete.author=None
+            themetodelete.save()
+        serialized_theme = ThemeSerializer(user.selected_theme, many=False)
+        themeOrderedDict = serialized_theme.data
+        return JsonResponse(themeOrderedDict, safe=False, status=200)
+        
 
 @api_login_required()
-def themeList(request):
+def additionalthemeList(request):
+    #this functon list out the additional themes
+    user = request.user
     if request.method == "GET":
-        themes = Theme.objects.all()
+        themes = Theme.objects.exclude(author=user).exclude(name__in=["Daily Bruin", "Dark Bruin"]).order_by('-favorite_count')
         serialized_themes = ThemeSerializer(themes, many=True)
         themeOrderedDict = serialized_themes.data
-        return JsonResponse(themeOrderedDict, safe=False)
+        starred_themes_id =[theme.pk for theme in user.starred_themes.all()] 
+        return JsonResponse({'additionalThemes': themeOrderedDict, 'starredThemesId': starred_themes_id }, safe=False, status=200)
+
+@api_login_required()
+def themeStar(request):
+    user = request.user
+    if request.method == "POST":
+        req_data = json.loads(request.body)
+        theme_id = req_data['id']
+        themes = Theme.objects.all()
+        star_theme = themes.get(pk=theme_id)
+        user.starred_themes.add(star_theme)
+        starred_themes_id = user.starred_themes.values_list('pk', flat=True)
+        starred_themes_id = list(starred_themes_id)
+        return JsonResponse({'starredThemesId': starred_themes_id, 'favCount': themes.get(pk=theme_id).favorite_count }, safe=False, status=200)
+    elif request.method == "PUT":
+        req_data = json.loads(request.body)
+        theme_id = req_data['id']
+        themes = Theme.objects.all()
+        #theme we want to remove the relation
+        unstar_theme = themes.get(pk=theme_id)
+        user.starred_themes.remove(unstar_theme)
+        if unstar_theme.author==None and unstar_theme.user_set.count()==0 and unstar_theme.related_users.count()==1:
```

Comment:
Just for safety, we should check that the `unstar_theme` is not Daily Bruin or Dark Bruin

Reply:
Additionally, this functionality is used in multiple places. We should put this as a method inside the Theme model. That way you can say, `unstar_theme.delete_if_unused()`.

Reply:
i added a function called check_if_unused which i will call and it will return a boolean if true then delete

Reply:
the deletion is done in the views

Reply:
fixed it according to our discussion on slack

Reply:
fixed it according to our discussion on slack